src/modules/java.rs

```diff
@@ -0,0 +1,71 @@
+use std::process::Command;
+
+use ansi_term::Color;
+
+use super::{Context, Module};
+
+/// Creates a module with the current Java version
+///
+/// Will display the Java version if any of the following criteria are met:
+///     - Current directory contains a file with a `.java` extension
+///     - Current directory contains a `pom.xml` file
+pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
+    let is_java_project = context
+        .new_scan_dir()
+        .set_files(&["pom.xml"])
+        .set_extensions(&["java"])
+        .scan();
+
+    if !is_java_project {
+        return None;
+    }
+
+    match get_java_version() {
+        Some(java_version) => {
+            const JAVA_CHAR: &str = "☕ ";
+            let module_color = Color::RGB(166, 42, 42).bold();
+
+            let mut module = context.new_module("java")?;
+            module.set_style(module_color);
+
+            let formatted_version = format_java_version(java_version);
+            module.new_segment("symbol", JAVA_CHAR);
+            module.new_segment("version", &formatted_version);
+
+            Some(module)
+        }
+        None => None,
+    }
+}
+
+fn get_java_version() -> Option<String> {
+    match Command::new("javac").arg("-version").output() {
+        Ok(output) => {
+            if output.stdout.is_empty() {
```

Comment:
Whoa, do some versions of javac print the version info to stderr?

Reply:
As far as I can tell, it only prints to stderr, but haven't found any definitive answer for this, which is why I included both.

Reply:
Alright, good to know! Could you leave a comment here about that so that we know why we're reading from both input streams?